src/psyclone/tests/domain/common/algorithm/alg_psyir_test.py

```diff
@@ -0,0 +1,190 @@
+# Author R. W. Ford, STFC Daresbury Lab
+
+'''Module containing tests for the translation of PSyIR to PSyclone
+Algorithm PSyIR.
+
+'''
+from __future__ import absolute_import
+import pytest
+
+from psyclone.psyir.nodes import Reference, Node
+from psyclone.psyir.nodes.node import colored
+from psyclone.psyir.symbols import RoutineSymbol, TypeSymbol, \
+    StructureType, Symbol
+from psyclone.domain.common.algorithm import AlgorithmInvokeCall, \
+    KernelFunctor
+from psyclone.errors import GenerationError
+
+
+def test_algorithminvokecall():
+    '''Check that an instance of AlgorithmInvokeCall can be
+    created.
+
+    '''
+    routine = RoutineSymbol("hello")
+    call = AlgorithmInvokeCall(routine)
+    assert call._text_name == "AlgorithmInvokeCall"
+    assert call._colour == "green"
+
+
+def test_algorithminvokecall_invalid_arg():
+    '''Check that the create method of an AlgorithmInvokeCall raises the
+    expected exception if its supplied children are not the expected
+    type (tests _validate_child method and _children_valid_format
+    variable).
+
+    '''
+    routine = RoutineSymbol("symbol")
+    with pytest.raises(GenerationError) as info:
+        _ = AlgorithmInvokeCall.create(routine, ["hello"])
+    assert("Item 'str' can't be child 0 of 'AlgorithmInvokeCall'. The valid "
+           "format is: '[KernelFunctor]*'." in str(info.value))
+
+
+def test_kernellayerref():
+    '''Check that an instance of KernelFunctor class can be created. Also
+    check that the symbol method works as expected.
+
+    '''
+    symbol = TypeSymbol("hello", StructureType())
+    klr = KernelFunctor(symbol)
+    assert klr._symbol == symbol
+    assert klr.symbol == symbol
+    assert klr._colour == "yellow"
+    assert klr._text_name == "KernelFunctor"
+    assert klr.parent is None
+
+
+def test_kernellayerref_parent():
+    '''Check that the optional parent argument to a KernelFunctor class
+    constructor is stored correctly.
+
+    '''
+    parent = Node()
+    symbol = TypeSymbol("hello", StructureType())
+    klr = KernelFunctor(symbol, parent=parent)
+    assert klr.parent == parent
+
+
+def test_kernellayerref_invalid_symbol():
+    '''Check KernelFunctor raises the expected exception if the type of
+    the symbol argument is invalid.
+
+    '''
+    with pytest.raises(TypeError) as info:
+        _ = KernelFunctor(Symbol("hello"))
+    assert ("KernelFunctor symbol argument should be a TypeSymbol but "
+            "found 'Symbol'." in str(info.value))
+
+
+class SubClass(KernelFunctor):
+    '''Utility subclass of KernelFunctor to test that the create method
+    in KernelFunctor behaves as expected for subclasses.'''
+
+
+@pytest.mark.parametrize("cls", [KernelFunctor, SubClass])
+def test_kernellayerref_create(cls):
+    '''Check that the create method of KernelFunctor works as expected.
+
+    '''
+    symbol = TypeSymbol("hello", StructureType())
+    klr = cls.create(symbol, [])
+    assert type(klr) is cls
```

Comment:
Pylint says unidiomatic-typecheck. Can this be an isinstance, otherwise add the pylint disable statement

Reply:
This needs to check the specific class. I've added the pylint disable directive.